### PR TITLE
Handling links in chapter help text

### PR DESCRIPTION
**Pull request by qjac**
Closes https://github.com/GSA/openacr/issues/288 and closes https://github.com/GSA/openacr/issues/290

- Created a function to open first accordion for anchor links in chapter help text.
- Stopped the alert (You are about to lose content) from showing when users click on the internal, but different route link in the 'software' help text.

## QA
Note: Test in multiple browsers and please comment re: which browsers tested.

- After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
- Confirm that chapter help text is displaying properly (correct text for the correct chapter and is well-formatted)
- For A and AA, test the links in the chapter help text. Confirm that if a link is clicked, the first accordion on the page opens. 
- For Software, click the link in the chapter help text. Confirm that you navigate to A without getting the alert. 
- Return to the 'Software' page and modify an input field. Then click the link. Confirm that you navigate to A without getting the alert. Next navigate to an external url, confirm that you DO get the alert at that point

### CODE REVIEW

File: src/components/ChapterHelpText.svelte [hunk 1]
```diff
@@ -1,8 +1,77 @@
...
+      section may be used to summarize accessibility features described in more
+      detail elsewhere.
+    {:else if chapterId === "hardware"}
+      If the Information and Communications Technology (ICT) includes hardware,
+      please fill out this section.
+    {:else if chapterId === "software"}
+      If your software product is a web application or is rendered through a web
+      browser, do not fill out this section. Fill out <a
+        href="success_criteria_level_a"
+        on:click={() => (window.onbeforeunload = null)}>Tables 1, 2, and 3</a
```
Comment (dmundra) [inline src/components/ChapterHelpText.svelte:66]: Nice solution here. I was going to suggest the `<Link>` component but this is better example to follow.

Comment (dmundra) [reply src/components/ChapterHelpText.svelte:66]: @qjac minor nitpick here. When I test it, it resets the warning. That is if I change a form input (check a box or type something) then I click that link then click the GSA logo/external link it doesn't give the warning. What do you think?

**Review by dmundra — APPROVED** [review]
Looking good! Tested on Mac and these browsers:

- Chrome 97.0.4692.99 (Official Build) (x86_64)
- Firefox 96.0.2
- Edge 97.0.1072.69 😝 
- Safari 15.2

**Comment by danid123** [comment]
Looks good to me, just a small nit pick. 

In the help text we list the sections in this order: 
"The default is web-based, and there are optional fields for Documentation, Authoring and Software"

But they appear in Documentation, Software, Authoring order so the sentence should probably read:
"The default is web-based, and there are optional fields for Documentation, Software, and Authoring."

The change would apply to Table 1 and Table 2.

@mgifford is there a reason why we don't also refer to the Web section in the help text?

I tested on a Mac in:

- Chrome 96.0.4664.110 (haven't updated yet)
- Firefox 96.0.2
- Safari 15.2

**Review by danid123 — APPROVED** [review]

File: src/components/ChapterHelpText.svelte (hunk 1 above)
Comment (qjac) [reply src/components/ChapterHelpText.svelte:66]: Primary importance is to not lose data... I'll do some more exploring/testing.

Comment (qjac) [reply src/components/ChapterHelpText.svelte]: Full disclosure, I do not know why <Link> would not work. Nor do I really understand why use:link works. But the functionality does seem to fix, the issue with allowing subsequent external links to load without the warning.

Comment (dmundra) [reply src/components/ChapterHelpText.svelte]: Looks like `use:link` is a fancy Svelte navigator feature https://github.com/mefechoel/svelte-navigator#link-1, beats me as well but hey it works! Lets run with it.

**Review by dmundra — APPROVED** [review]
Thank you for making the changes!

**Pull request merged**